src/applet.rs

```diff
@@ -35,22 +45,34 @@ pub struct SystemMonitorApplet {
     nets: Networks,
     disks: Disks,
     gpus: Gpus,
+    /// Component for handling CPU thermal sensor data. Added for CPU temperature monitoring.
+    thermal: CpuThermalSensors,
     /// percentage global cpu used between refreshes
     global_cpu: History<f32>,
     ram: History,
     swap: History,
-    /// amount uploaded between refresh of `sysinfo::Nets`. (DOES NOT STORE RATE)
-    upload: History,
-    /// amount downloaded between refresh of `sysinfo::Nets`. (DOES NOT STORE RATE)
-    download: History,
-    /// amount read between refresh of `sysinfo::Disks`. (DOES NOT STORE RATE)
-    disk_read: History,
-    /// amount written between refresh of `sysinfo::Disks`. (DOES NOT STORE RATE)
-    disk_write: History,
-    /// amount read between refresh of `sysinfo::Disks`. (DOES NOT STORE RATE)
+    /// History of network upload rate in MB/s. Changed from u64 (bytes) to f32 (MB/s).
+    upload: History<f32>,
+    /// History of network download rate in MB/s. Changed from u64 (bytes) to f32 (MB/s).
+    download: History<f32>,
+    /// History of disk read rate in MB/s. Changed from u64 (bytes) to f32 (MB/s).
+    disk_read: History<f32>,
+    /// History of disk write rate in MB/s. Changed from u64 (bytes) to f32 (MB/s).
+    disk_write: History<f32>,
+    /// GPU usage history for each GPU.
```

Comment:
Why did you change to `f32`? This reduces precision compared to `u64`.

If anything, the fix of an eventual overflow of EXABYTES of I/O is to use wrapping subtraction. I don't think anyone with a normal PC is able to use that much.